api/v1alpha1/controlplane_types.go

```diff
@@ -90,12 +107,29 @@ type SecretReference struct {
 	// `name` is the name of the secret.
 	// Required
 	Name string `json:"name"`
-	// Required
+	// This field is present for control planes of type `k8s`, `vcluster`, `ocm`, `host`.`
+	// it is not present for control planes of type `external`.
+	// Controllers for control planes of type `external` should always use the `InClusterKey`.
```

Comment:
That implies that such controllers, and indeed any client, has to run in the KubeFlex hosting cluster. That seems surprisingly and unnecessarily restrictive to me; the other types of ControlPlane do not have such a restriction.